src/lib/storage/proposal/partition_creator.rb

```diff
@@ -281,14 +281,27 @@ def create_partition(vol, partition_id, free_slot)
           log.info("Creating partition for #{vol.mount_point} with #{vol.size}")
           disk = ::Storage::Disk.find(devicegraph, free_slot.disk_name)
           ptable = disk.partition_table
-          if logical_partition_preferred?(ptable)
-            create_extended_partition(disk, free_slot.slot.region) unless ptable.has_extended
+
+          logical = false
+          if space_inside_extended?(free_slot)
+            logical = true
+          elsif logical_partition_preferred?(ptable)
+            # It's not possible to have more than one extended partition,
+            # even if logical partitions are preferred
+            if !ptable.has_extended
+              create_extended_partition(disk, free_slot.slot.region)
+              logical = true
+            end
+          end
```

Comment:
Uhm, it really **can** happen:
[ gap1, p1, p2, gap2 ] ->  add small partition ->
[p0, gap1, p1, p2, gap2] -> add small partition -> boom